common/copycleanuphandler.cpp

```diff
@@ -0,0 +1,34 @@
+
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "copycleanuphandler.h"
+
+#include "recursivecopy.h"
+
+CopyCleanupHandler::CopyCleanupHandler(std::string src, std::string dst)
+{
+    m_src = src;
+    m_dst = dst;
+}
+
+ReturnCode CopyCleanupHandler::clean()
+{
+    return RecursiveCopy::getInstance().copy(m_src, m_dst);
```

Comment:
It feels counter-intuitive (when no docs are given in the .h file) that the clean step copies something. I think the naming of this is a bit strange.